flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1031,8 +1036,10 @@ protected void closeJobManagerConnection(JobID jobId, Exception cause) {
 
             jmResourceIdRegistrations.remove(jobManagerResourceId);
 
-            slotManager.processResourceRequirements(
-                    ResourceRequirements.empty(jobId, jobMasterGateway.getAddress()));
+            if (resetRequirements) {
+                slotManager.processResourceRequirements(
```

Comment:
if we want to retain this safeguard for all cases we could also forward the flag to the slot manager, indicating whether slots should be released or not.
It's a bit jank that resetting the requirements is now coupled with releasing inactive slots...

Reply:
but maybe that is fine; the release of inactive slots is just a detail of the declarative slot protocol that the RM doesn't need to know about.

Reply:
I think this is fine from my perspective.